packages/dashboard/src/store/actions/pasteWidgets/index.ts

```diff
@@ -37,18 +42,43 @@ export const pasteWidgets = (state: DashboardState, action: PasteWidgetsAction):
       y: position && Math.floor(position.y / cellSize),
     };
 
-    const widgetToCompare = first(sortBy(copyGroup, ['x', 'y']));
+    const leftmostWidget: DashboardWidget = minBy(copyGroup, 'x') || copyGroup[0];
+    const groupLeftX = leftmostWidget.x;
+
+    const topmostWidget: DashboardWidget = minBy(copyGroup, 'y') || copyGroup[0];
+    const groupTopY = topmostWidget.y;
+
+    const widgetsRightPos = map(copyGroup, (widget) => {
+      return widget.x + widget.width;
+    });
+    const groupRightX = max(widgetsRightPos) || xgrid;
+
+    const widgetsBottomPos = map(copyGroup, (widget) => {
+      return widget.y + widget.height;
+    });
+    const groupBottomY = max(widgetsBottomPos) || ygrid;
+
+    const groupWidth = groupRightX - groupLeftX;
+    const groupHeight = groupBottomY - groupTopY;
+
     offset = {
-      x: cellPosition.x - (widgetToCompare?.x ?? 0),
-      y: cellPosition.y - (widgetToCompare?.y ?? 0),
+      x: cellPosition.x - groupLeftX,
+      y: cellPosition.y - groupTopY,
     };
+
+    if (cellPosition.x + groupWidth > xgrid) {
```

Comment:
I think the logic makes sense, can you add some comments so that when someone is looking at this,  it makes it a bit easier to understand?